### PR TITLE
[KOGITO-9319] OpenVino example

### PR DESCRIPTION
**Pull request by fjtirado**
OpenVino [notebook helloword](https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/001-hello-world) rewritten as serverless workflow. 

Depends on https://github.com/kiegroup/kogito-runtimes/pull/3090 for native mode

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - serverless-workflow-examples job** `#280` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT= 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-examples -u https://github.com/kiegroup/kogito-examples/pull/1698 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-examples.tests.serverless-workflow-examples/280/display/redirect

**Test results:**
- PASSED: 75
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-examples.tests.serverless-workflow-examples/280/testReport/org.acme.serverless.loanbroker.flow/LoanBrokerFlowTest/verifyLoanBroker/">org.acme.serverless.loanbroker.flow.LoanBrokerFlowTest.verifyLoanBroker</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor#setupKubernetesDevService threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed<br/> at io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor.setupKubernetesDevService(DevServicesKubernetesProcessor.java:119)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:281)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)<br/> at java.base/java.lang.Thread.run(Thread.java:829)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:349)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)<br/> at io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor.lambda$startKubernetes$7(DevServicesKubernetesProcessor.java:227)<br/> at java.base/java.util.Optional.orElseGet(Optional.java:369)<br/> at io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor.startKubernetes(DevServicesKubernetesProcessor.java:241)<br/> at io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor.setupKubernetesDevService(DevServicesKubernetesProcessor.java:109)<br/> ... 11 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)<br/> ... 16 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:542)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 17 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:349)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)<br/> at com.dajudge.kindcontainer.ApiServerContainer.containerIsStarting(ApiServerContainer.java:142)<br/> at org.testcontainers.containers.GenericContainer.containerIsStarting(GenericContainer.java:692)<br/> at com.dajudge.kindcontainer.KubernetesContainer.containerIsStarting(KubernetesContainer.java:151)<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:474)<br/> ... 19 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)<br/> ... 24 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:542)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 25 more<br/>Caused by: com.github.dockerjava.api.exception.ConflictException: Status 409: {"message":"cannot join network of a non running container: d79d4eef97209bce73cd6b53a6fb56dbacb6ea5cba021f25a130c36ce37937f6"}<br/><br/> at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:245)<br/> at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:102)<br/> at org.testcontainers.shaded.com.github.dockerjava.core.exec.StartContainerCmdExec.execute(StartContainerCmdExec.java:31)<br/> at org.testcontainers.shaded.com.github.dockerjava.core.exec.StartContainerCmdExec.execute(StartContainerCmdExec.java:13)<br/> at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)<br/> at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)<br/> at org.testcontainers.shaded.com.github.dockerjava.core.command.StartContainerCmdImpl.exec(StartContainerCmdImpl.java:43)<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:441)<br/> ... 27 more<br/>
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-examples.tests.serverless-workflow-examples/280/testReport/org.kie.kogito.examples/TimeoutIT/Build_projects___testSwitchStateWorkflow/">Build projects / org.kie.kogito.examples.TimeoutIT.testSwitchStateWorkflow</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>

**Comment by kie-ci3** [comment]
**(tests) - serverless-workflow-examples job** `#281` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT= 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-examples -u https://github.com/kiegroup/kogito-examples/pull/1698 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-examples.tests.serverless-workflow-examples/281/display/redirect

**Test results:**
- PASSED: 78
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-examples.tests.serverless-workflow-examples/281/testReport/org.acme.newsletter.subscription.service/SubscriptionResourceIT/verifySubscriptionCycle/">org.acme.newsletter.subscription.service.SubscriptionResourceIT.verifySubscriptionCycle</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor#launchDatabases threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed<br/> at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(DevServicesDatasourceProcessor.java:345)<br/> at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.launchDatabases(DevServicesDatasourceProcessor.java:136)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:281)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)<br/> at java.base/java.lang.Thread.run(Thread.java:829)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:349)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)<br/> at io.quarkus.devservices.postgresql.deployment.PostgresqlDevServicesProcessor$1.startDatabase(PostgresqlDevServicesProcessor.java:65)<br/> at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(DevServicesDatasourceProcessor.java:282)<br/> ... 12 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)<br/> ... 15 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:542)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 16 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*database system is ready to accept connections.*\s'<br/> at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)<br/> at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)<br/> at org.testcontainers.containers.PostgreSQLContainer.waitUntilContainerStarted(PostgreSQLContainer.java:143)<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:485)<br/> ... 18 more<br/>
</details>

**Comment by fjtirado** [comment]
jenkins run serverless-workflow-examples tests

**Review by ricardozanini — APPROVED** [review]
Pretty cool!

File: serverless-workflow-examples/serverless-workflow-openvino-quarkus/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Kogito Serverless Workflow - Openvino Hello World Example
+
+## Description
+
+This example contains a workflow definition that emulates functionality exposed by [openvino hello world example](https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/001-hello-world/001-hello-world.ipynb)
+
+The flow, given a image file name containing a dog image, returns that dog race.
```
Comment (MarianMacik) [inline serverless-workflow-examples/serverless-workflow-openvino-quarkus/README.md]: I don't want to be nitpicking, no worries, I don't check the grammar, just noticed it while reading :)
```suggestion
The flow, given an image file name containing a dog image, returns that dog's race. 
```

File: serverless-workflow-examples/serverless-workflow-openvino-quarkus/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+### Compile and Run in JVM mode
+
+```sh
+mvn clean package 
+java -jar target/quarkus-app/quarkus-run.jar
+```
+
+or on windows
```
Comment (MarianMacik) [inline serverless-workflow-examples/serverless-workflow-openvino-quarkus/README.md]:
```suggestion
or on Windows
```

File: serverless-workflow-examples/serverless-workflow-openvino-quarkus/pom.xml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.kie.kogito.examples</groupId>
+  <artifactId>serverless-workflow-openvino-quarkus</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>Kogito Example :: Serverless Workflow Openvino :: Quarkus</name>
+  <description>Kogito Serverless Workflow OpenVino Example - Quarkus</description>
+  <properties>
+    <quarkus-plugin.version>2.16.6.Final</quarkus-plugin.version>
```
Comment (MarianMacik) [inline serverless-workflow-examples/serverless-workflow-openvino-quarkus/pom.xml]: We're on 2.16.7.Final already.
```suggestion
    <quarkus-plugin.version>2.16.7.Final</quarkus-plugin.version>
```

File: serverless-workflow-examples/serverless-workflow-openvino-quarkus/pom.xml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
+    <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
+    <quarkus.platform.version>2.16.6.Final</quarkus.platform.version>
```
Comment (MarianMacik) [inline serverless-workflow-examples/serverless-workflow-openvino-quarkus/pom.xml]:
```suggestion
    <quarkus.platform.version>2.16.7.Final</quarkus.platform.version>
```

File: serverless-workflow-examples/serverless-workflow-openvino-quarkus/src/main/resources/openvino.sw.json [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+              "script": "from pathlib import Path\nimport sys\nimport cv2\nimport matplotlib.pyplot as plt\nimport numpy as np\nfrom openvino.runtime import Core\nsys.path.append('./utils')\nfrom notebook_utils import download_file"
+            }
+          }
+        },
+        {
+          "functionRef": {
+            "name":"Model names",
+            "refName": "python",
+            "arguments": {
+              "script": "base_artifacts_dir = Path('./model').expanduser()\nmodel_name = 'v3-small_224_1.0_float'\nmodel_xml_name = f'{model_name}.xml'\nmodel_bin_name = f'{model_name}.bin'\nmodel_xml_path = base_artifacts_dir / model_xml_name\nase_url = 'https://storage.openvinotoolkit.org/repositories/openvino_notebooks/models/mobelinet-v3-tf/FP32/'"
```
Comment (MarianMacik) [inline serverless-workflow-examples/serverless-workflow-openvino-quarkus/src/main/resources/openvino.sw.json]: Shouldn't it be?
```suggestion
              "script": "base_artifacts_dir = Path('./model').expanduser()\nmodel_name = 'v3-small_224_1.0_float'\nmodel_xml_name = f'{model_name}.xml'\nmodel_bin_name = f'{model_name}.bin'\nmodel_xml_path = base_artifacts_dir / model_xml_name\nbase_url = 'https://storage.openvinotoolkit.org/repositories/openvino_notebooks/models/mobelinet-v3-tf/FP32/'"
```

**Review by MarianMacik — APPROVED** [review]
That's very cool, great job @fjtirado!
Just a few things, nothing major.

**Pull request merged**